aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1911,7 +1927,10 @@ impl AptosSimulationVM {
 
         // Revalidate the transaction.
         let txn_data = TransactionMetadata::new(txn);
-        let mut session = self.0.new_session(resolver, SessionId::txn_meta(&txn_data));
+        let mut session = self
+            .0
+             .0
```

Comment:
both of these 0's are rly annoying to look at

Reply:
They are, doing that now in a separate PR :)

Reply:
have a look at https://github.com/aptos-labs/aptos-core/pull/8220, was an attempt to do some things you are addressing now.

Reply:
Nice, i remember Gerben tried😂 let's do it :)

Reply:
Btw, this is simulation, so it will go away but first I need another PR so that refactoring and simulation are separeted